mindsdb/integrations/utilities/infer_types.py

```diff
@@ -0,0 +1,54 @@
+import pandas as pd
+
+
+def infer_column_type(column, type=None):
```

Comment:
I feel like we should drop these methods, and use `type_infer` instead. If it does not work correctly, then frame it as "what can/should we add from this method into that package", as anything having to do with type inference for ML (like this) should go into that package.

Reply:
What functionality is required now in timegpt handler:
- convert one column in datetime if it possible.

I could replace it with using of type_infer module like this:
```python

sample_data = df[column].sample(min(100, len(df)))
if type_infer.type_check_date(sample_data):
    df[column] = pd.to_datetime(df[column])
```

Would it be better? Or there could be another more preferable solution?

Reply:
done

Reply:
I think there's a better way (also, the learning hub example does not work as-is). 

Here is a set of changes that work, we may want to merge that [here](https://github.com/mindsdb/mindsdb/compare/fix-timegpt...fix_timegpt_7419_v2).